### ISSUE TITLE
Style/HashSyntax still consider not safe when autocorrect with -a

### ISSUE THREAD

**Issue by zw963** [rubocop/rubocop#10545]
When we have a following file.

```rb
def foo(operator)
  send_notify users, operator: operator
end
```

It will be autocorrect to following code when save.

```rb
def foo(operator)
  send_notify users, operator:
end
```

Yes, it correct, no syntax error, but when we add a line code after it, like this:

```rb
def foo(operator)
  send_notify users, operator:
                       puts 'ok'                   # <= indent error, in fact, syntax too.
end
```

As you can see, `puts 'ok'` even can not be indent correctly by emacs builtin `ruby-mode` on emacs 28.1
That is because this cause syntax error:

```
syntax error, unexpected string literal, expecting `do' or '{' or '('
                       puts 'ok'
```

so, i consider Style/HashSyntax should not do above changes, another solution is, add parenthesis too. like this:

```rb
def foo(operator)
  send_notify(users, operator:)
  puts 'ok'                    
end
```

thank you.

rubocop 1.27.0
ruby 3.1.0

**Comment by brian-kephart** [comment]
I came here to file the same issue. This cop is performing unsafe autocorrection on arguments that are not wrapped in parentheses. This happened to me in real-world code such as the following:

```ruby
get enrollments_path start_date: start_date, end_date: end_date
assert_response :ok
```

Which autocorrected to:
```ruby
get enrollments_path start_date:, end_date:
assert_response :ok

#=> syntax error, unexpected symbol literal, expecting `do' or '{' or '(' (SyntaxError)
#=> assert_response :ok
#=>                 ^
```

RuboCop 1.36.0
Ruby 3.1.2

**Comment by koic** [comment]
> As you can see, puts 'ok' even can not be indent correctly by emacs builtin ruby-mode on emacs 28.1
> That is because this cause syntax error:

Hm, this is related to the following current Ruby syntax issue:
https://bugs.ruby-lang.org/issues/18396

Maybe, an ideal solution would be for ruby-mode to be able to make sense of correct indentation to the hash omission.

> ```ruby
> get enrollments_path start_date: start_date, end_date: end_date
> assert_response :ok
> ```

By updating to the latest RuboCop (currently 1.41.1) , auto-correction will add parenthesis.

```ruby
get enrollments_path(start_date:, end_date:)
assert_response :ok
```
